com.onesignal.unity.ios/Runtime/iOSNotificationsManager.cs

```diff
@@ -48,15 +48,28 @@ internal sealed class iOSNotificationsManager : INotificationsManager {
         [DllImport("__Internal")] private static extern void _notificationsWillDisplayEventPreventDefault(string notificationId);
         [DllImport("__Internal")] private static extern void _notificationsSetClickCallback(ClickListenerDelegate callback);
 
-        public delegate void PermissionListenerDelegate(bool permission);
+        private delegate void PermissionListenerDelegate(bool permission);
         private delegate void WillDisplayListenerDelegate(string notification);
         private delegate void ClickListenerDelegate(string notification, string resultActionId, string resultUrl);
         private delegate void BooleanResponseDelegate(int hashCode, bool response);
 
         public event EventHandler<NotificationWillDisplayEventArgs> ForegroundWillDisplay;
-        public event EventHandler<NotificationClickEventArgs> Clicked;
         public event EventHandler<NotificationPermissionChangedEventArgs> PermissionChanged;
 
+        private bool _clickDelegate;
+        private EventHandler<NotificationClickEventArgs> _clicked;
+        public event EventHandler<NotificationClickEventArgs> Clicked {
+            add {
+                // only add one click listener
+                if (!_clickDelegate) {
+                    _clicked += value;
```

Comment:
We want to allow the developer to add more than one click listener, so this line should be outside of the if statement.